dissect/target/plugins/os/windows/_os.py

```diff
@@ -110,24 +112,153 @@ def ips(self) -> list[str]:
+
+        Returns:
+            The integer value of the NT major version number or ``None`` if the
+            CurrentMajorVersionNumber sub key is not present or any other
+            registry error occurred.
+        """
+        return self._get_version_reg_value("CurrentMajorVersionNumber")
+
+    def _minor_version(self) -> Optional[int]:
+        """Return the NT minor version number (starting from NT 10.0 / Windows 10).
+
+        Returns:
+            The integer value of the NT minor version number or ``None`` if the
+            CurrentMinorVersionNumber sub key is not present or any other
+            registry error occurred.
+        """
+        return self._get_version_reg_value("CurrentMinorVersionNumber")
+
+    def _nt_version(self) -> Optional[int]:
+        """Return the Windows NT version in x.y format.
+
+        For systems up to and including NT 6.3 (Win 8 / Win 2012 Server) this
+        will be the value of the CurrentVersion sub key.
+
+        For systems starting from NT 10.0 (Win 10) this will be a value
+        constructed from the combination of the CurrentMajorVersionNumber and
+        CurrentMinorVersionNumber sub keys.
+
+        Returns:
+            The string value of the NT version or ``None`` if the any of the
+            sub keys are not present or any other registry error occurred.
+        """
+        version = None
+
+        major_version = self._major_version()
+        if major_version is not None:
+            minor_version = self._minor_version()
+            if minor_version is None:
+                minor_version = ""
+            version = f"{major_version}.{minor_version}"
+        else:
+            version = self._legacy_current_version()
+
+        return version
+
     @export(property=True)
     def version(self) -> Optional[str]:
-        key = "HKLM\\Software\\Microsoft\\Windows NT\\CurrentVersion"
-        csd_version = str()
+        """Return a string representation of the Windows version of the target.
 
-        try:
-            csd_version = self.target.registry.key(key).value("CSDVersion").value
-        except RegistryError:
-            pass
+        The version string will look like:
 
-        try:
-            r = self.target.registry.key(key)
-            product_name = r.value("ProductName").value
-            current_version = r.value("CurrentVersion").value
-            current_build_number = r.value("CurrentBuildNumber").value
-            return f"{product_name} (NT {current_version}) {current_build_number} {csd_version}"
-        except RegistryError:
-            pass
+        for windows versions until Windows 10:
+        ``<ProductName> (NT <CurrentVersion>) <CurrentBuildNumber>.<UBR> <CSDVersion>``
+
+        for windows versions starting from Windows 10:
+
+        ``<ProductName> (NT
+        <CurrentMajorVersionNumber>.<CurrentMinorVersionNumber>) \
+        <CurrentBuildNumber>.<UBR> <CSDVersion>``
```

Comment:
```
"""Return a string representation of the Windows version of the target.

For Windows versions before Windows 10 this looks like::

    <ProductName> (NT <CurrentVersion>) <CurrentBuildNumber>.<UBR> <CSDVersion>

For Windows versions after Windows 10 it looks like::

    <ProductName> (NT <CurrentMajorVersionNumber>.<CurrentMinorVersionNumber>) <CurrentBuildNumber>.<UBR> <CSDVersion>
```
Personally I'd leave it as a long line and put a `# noqa` on the docstring, makes it easier to read.